src/GNDStk/gnds-tree-node.hpp

```diff
@@ -0,0 +1,444 @@
+      // Otherwise, we'll assume that the caller intends to deal with the issue
+      // in its own way, and therefore that we shouldn't produce such an error.
+      // Note that the question of whether a "found" flag was sent is determined
+      // by looking at its address. This is entirely different from the question
+      // of what found's value should be. Its value is set either way.
+      if (&found == &static_found) {
+         // found references our private default - so, caller didn't send one
+         error(
+            "Node meta() called with key \"" + str + "\", "
+            "but this key wasn't\nfound in the node's metadata."
+         );
+      }
+
+      // done
+      return empty;
+   }
+
+   // for meta_t<T>
+   // Return by value isn't ideal, if T is something large like a container.
+   // Think about options.
+   template<class T>
+   T meta(
+      const meta_t<T> &kwd,
+      bool &found = static_found
+   ) const {
+      const std::string &str = meta(kwd.name,found);
+      T value{};
+      if (found)
+         gnds::read(str,value);
+      return value;
+   }
+
+   // for meta_t<string>
+   // Functionally equivalent to using meta(meta_t<T>) with T = string,
+   // but more direct and thus perhaps more efficient.
+   std::string meta(
+      const meta_t<std::string> &kwd,
+      bool &found = static_found
+   ) const {
+      return meta(kwd.name,found); // direct call to meta(string)
+   }
+
+   // for meta_t<void>
+   // Let's define the meta_t<void> case to be equivalent to the meta_t<string>
+   // case. This makes the behavior of meta_t be more consistent with that of
+   // child_t, which uses <void> to stipulate that the child node be returned
+   // in its original tree-node form.
+   std::string meta(
+      const meta_t<void> &kwd,
+      bool &found = static_found
+   ) const {
+      return meta(kwd.name,found); // as above
+   }
+
+   // for meta_t<variant>, caller must stipulate <T>.
+   // We can't just fold this into meta(meta_t<T>) above and return the variant,
+   // because the read() would have no idea what variant variation to read into!
+   template<class T, class... Ts>
+   T meta(
+      const meta_t<std::variant<Ts...>> &kwd,
+      bool &found = static_found
+   ) const {
+      // body is as above, but the function signature is structurally different
+      const std::string &str = meta(kwd.name,found);
+      T value{}; // T having been direct-specified, not in meta_t<here>
+      if (found)
+         gnds::read(str,value);
+      return value;
+   }
+
+
+   // ------------------------
+   // child()
+   // ------------------------
+
+   // for string
+   const Node &child(
+      const std::string &str,
+      bool &found = static_found // as for meta(string)
+      // fixme Need "found" below, and in operator()
+   ) const {
+      // search
+      for (auto &c : children)
+         if (c != nullptr && c->name == str)
+            return found = true, *c;
```

Comment:
:-D Every once in a while, I'm in the mood to have a bit of fun with the language, but I'll only do things like this in a very few select cases. Someone who learns x = x+1 will find x += 1 to be disconcerting (for a short while). Someone who learns x += 1 will find x++ to be disconcerting (for a short while). If I write:

```
   return A, B, C, D, E;
```

It's equivalent to:

```
   {
   A;
   B;
   C;
   D;
   return E;
   }
```

While most people (including me) would regard the second, longer syntax as being clearer in most cases, I'll occasionally have a situation which could be described as, "just set some value quickly and then immediately return." Like, here, "flag that we found it, then return it." All in one fell swoop, so we don't need two lines, and don't need {...}. Less clutter, more clarity - at least if you "get" the idiom in question. It's a very C/C++-ish thing to do, and, like I said, I'm selective about where I do this sort of thing. If you really want me to change it, I will.

Reply:
So are we returning `found` and `*c`?

Reply:
We're setting found to true, and then returning *c. found gets returned in the sense that it's a reference parameter to the function, so that the caller gets back that it's true.